runtime/queries/sourcepawn/highlights.scm

```diff
@@ -0,0 +1,308 @@
+
+(fixed_dimension) @punctuation.bracket
+
+(dimension) @punctuation.bracket
+
+(array_indexed_access) @punctuation.bracket
+
+(escape_sequence) @string.escape
+
+; Constructors
+(new_expression
+  class: (identifier) @type
+  arguments: (call_arguments) @constructor)
+
+; Methodmaps
+(methodmap) @type.definition
+
+(methodmap
+  name: (identifier) @type)
+
+(methodmap
+  inherits: (identifier) @type)
+
+(methodmap_method_constructor
+  name: (identifier) @constructor)
+
+(methodmap_method
+  name: (identifier) @function.method)
+
+(methodmap_native
+  name: (identifier) @function.method)
+
+(methodmap_property
+  name: (identifier) @property)
+
+(methodmap_property_getter) @function.method
+
+(methodmap_property_setter) @function.method
+
+; Enum structs
+(enum_struct) @type.definition
+
+(enum_struct
+  name: (identifier) @type)
+
+(enum_struct_field
+  name: (identifier) @variable.member)
+
+(enum_struct_method
+  name: (identifier) @function.method)
+
+; Non-type Keywords
+(variable_storage_class) @keyword.storage
+
+(visibility) @keyword.storage
+
+(visibility) @keyword.storage
+
+(assertion) @function.builtin
+
+(function_declaration_kind) @keyword.function
+
+[
+  "new"
+  "delete"
+] @keyword.operator
+
+[
+  "."
+  ","
+] @punctuation.delimiter
+
+; Operators
+[
+  "+"
+  "-"
+  "*"
+  "/"
+  "%"
+  "++"
+  "--"
+  "="
+  "+="
+  "-="
+  "*="
+  "/="
+  "=="
+  "!="
+  "<"
+  ">"
+  ">="
+  "<="
+  "!"
+  "&&"
+  "||"
+  "&"
+  "|"
+  "~"
+  "^"
+  "<<"
+  ">>"
+  ">>>"
+  "|="
+  "&="
+  "^="
+  "~="
+  "<<="
+  ">>="
+] @operator
+
+(ignore_argument) @operator
+
+(scope_access) @operator
+
+(rest_operator) @operator ; Should override (concatenated_string) but currently does nothing
+
+; public Plugin myinfo
+(struct_declaration
+  name: (identifier) @variable.builtin)
+
+; Typedef/Typedef
+(typeset) @type.definition
+
+(typedef) @type.definition
+
+(functag) @type.definition
+
+(funcenum) @type.definition
+
+(typedef_expression) @keyword.function ; function void(int x)
+
+; Enums
+(enum) @type.definition
+
+(enum
+  name: (identifier) @type)
+
+(enum_entry
+  name: (identifier) @constant)
+
+(enum_entry
+  value: (_) @constant)
+
+; Literals
+(int_literal) @number
+
+(char_literal) @character
+
+(float_literal) @number.float
+
+(string_literal) @string
+
+(array_literal) @punctuation.bracket
+
+(concatenated_string) @punctuation.delimiter
+
+(bool_literal) @boolean
+
+(null) @constant.builtin
+
+((identifier) @constant
+  (#eq? @constant "INVALID_HANDLE"))
+
+; Keywords
+"return" @keyword.return
+[
+  "if"
+  "else"
+  "case"
+  "default"
+  "switch"
+] @keyword.conditional
+[
+  "do"
+  "while"
+  "for"
+  "continue"
+  "break"
+] @keyword.repeat
+[
+  "__nullable__"
+  "delete"
+  "enum"
+  "funcenum"
+  "functag"
+  "get"
+  "methodmap"
+  "new"
+  "property"
+  "public"
+  "set"
+  "struct"
+  "typedef"
+  "typeset"
+  "void"
+] @keyword
+
+[
+  "const"
+  "native"
+  "static"
+  "stock"
+  "forward"
+] @type.qualifier
```

Comment:
The trailing newline is missing on a few files: `highlights.scm`, `injections.scm` and `languages.toml`